community/bolt/src/main/java/org/neo4j/bolt/logging/BoltMessageLog.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.neo4j.bolt.logging;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+
+import org.neo4j.io.ByteUnit;
+import org.neo4j.io.fs.FileSystemAbstraction;
+import org.neo4j.kernel.lifecycle.LifecycleAdapter;
+import org.neo4j.logging.FormattedLog;
+import org.neo4j.logging.Level;
+import org.neo4j.logging.Log;
+import org.neo4j.logging.RotatingFileOutputStreamSupplier;
+
+public class BoltMessageLog extends LifecycleAdapter
+{
+    private static final long ROTATION_THRESHOLD_BYTES = ByteUnit.MebiByte.toBytes( 20 );
+    private static final long ROTATION_DELAY_MS = TimeUnit.SECONDS.toMillis( 500 );
+    private static final int MAX_ARCHIVES = 10;
+
+    private final Log inner;
+
+    public BoltMessageLog( FileSystemAbstraction fileSystem, File logFile, Executor executor ) throws IOException
+    {
+        RotatingFileOutputStreamSupplier outputStreamSupplier = new RotatingFileOutputStreamSupplier( fileSystem,
```

Comment:
What file do these logs goto?

Reply:
They will go to file defined by `GraphDatabaseSettings.bolt_log_filename` setting. See `BoltMessageLogging` for more details.